### PR TITLE
r.geomorphon: Rewrite with parson and add json test

### LINKED ISSUE #6970: [Feat]  Rewrite r.geomorphon JSON output to use parson library
**Is your feature request related to a problem? Please describe.**
JSON output was added to r.geomorphon before introducing parson library.

The idea is to rewrite the current implementation to use parson library
for better maintainability.

### PR DESCRIPTION
**Pull request by saket0187**
This PR addresses #6970 

I have tried to write this in a cleaner way covering all edge cases. I have also added one test with nested loops to check each key-value pair (with floating-point handling). However, if it looks like too much, we can just use assertEqual. I have removed a few keys like timestamp and generator from expected output as they might cause problems.

Let me know if any changes are needed from my side or if tests need to be modified (or if expected values need to be simplified).

### CODE REVIEW

File: raster/r.geomorphon/profile.c [hunk 1]
```diff
@@ -251,45 +251,93 @@ static const char *format_token_common(const struct token *t)
...
-            continue;
-        case T_ESO:
-            if (indent == JSON_MIN_INDENT)
+        case T_SSO: {
+            G_JSON_Value *sub_val = G_json_value_init_object();
+            if (sub_val == NULL) {
+                G_fatal_error(
+                    _("Failed to initialize JSON object. Out of memory?"));
+                G_json_value_free(root_val);
                 return 0;
```
Comment (petrasovaa) [inline raster/r.geomorphon/profile.c]: This is dead code, fatal error exits

File: raster/r.geomorphon/testsuite/test_r_geom.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 from grass.gunittest.case import TestCase
 from grass.gunittest.main import test
 from grass.script.core import read_command
+import json
```
Comment (petrasovaa) [inline raster/r.geomorphon/testsuite/test_r_geom.py]: The import should be before the grass-specific imports, we don't enforce it but that's a python convention.

File: raster/r.geomorphon/testsuite/test_r_geom.py [hunk 3]
```diff
@@ -78,6 +296,49 @@ def test_sint(self):
         category = read_command("r.category", map=self.outsint)
         self.assertEqual(first=synth_out, second=category)
 
+    def test_json_profile(self):
+        """Test JSON profile output structure and key values"""
+
+        out = read_command(
```
Comment (petrasovaa) [inline raster/r.geomorphon/testsuite/test_r_geom.py]: let's use the Tools API

**Comment by petrasovaa** [comment]
When I run the new test without the change, I get:

```
======================================================================
FAIL: test_json_profile (__main__.TestClipling.test_json_profile)
Test JSON profile output structure and key values
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/akratoc/dev/grass/grass_main/raster/r.geomorphon/testsuite/test_r_geom.py", line 336, in test_json_profile
    self.assertAlmostEqual(
AssertionError: 133.6702117919922 != 133.67 within 4 places (0.00021179199220000555 difference)

----------------------------------------------------------------------
Ran 3 tests in 14.386s

FAILED (failures=1)

```
So the PR changes the way the floating point values are printed? It's not a big deal, but it would be good to clarify.

**Comment by saket0187** [comment]
> When I run the new test without the change, I get:
> 
> ```
> ======================================================================
> FAIL: test_json_profile (__main__.TestClipling.test_json_profile)
> Test JSON profile output structure and key values
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/home/akratoc/dev/grass/grass_main/raster/r.geomorphon/testsuite/test_r_geom.py", line 336, in test_json_profile
>     self.assertAlmostEqual(
> AssertionError: 133.6702117919922 != 133.67 within 4 places (0.00021179199220000555 difference)
> 
> ----------------------------------------------------------------------
> Ran 3 tests in 14.386s
> 
> FAILED (failures=1)
> ```
> 
> So the PR changes the way the floating point values are printed? It's not a big deal, but it would be good to clarify.

Yes, The old path went through format_token_common(), which used %.2f for T_MTR, so the number was rounded before it reached JSON.

**Review by petrasovaa — APPROVED** [review]
Thanks!
